framework/include/auxkernels/BuildArrayVariableAux.h

```diff
@@ -0,0 +1,28 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "AuxKernel.h"
+
+class BuildArrayVariableAux : public ArrayAuxKernel
+{
+public:
+  static InputParameters validParams();
+
+  BuildArrayVariableAux(const InputParameters & parameters);
+
+protected:
+  virtual void compute() override;
+
+  // ::computeValue is not used. The actual implementation is in ::compute.
+  virtual RealEigenVector computeValue() override { return RealEigenVector::Zero(_var.count()); }
```

Comment:
The fist final may not be needed. For example if later people want to add scaling coefficients for the composing variables, they will need to override `compute`.

Reply:
If this is the case - I would actually prefer that someone modify the base class with a scaling factor, because otherwise they'd duplicate all of the code in compute. 

We force people to not duplicate code by using final.